source/common/http/utility.cc

```diff
@@ -758,6 +758,11 @@ Utility::AuthorityAttributes Utility::parseAuthority(absl::string_view host) {
   const auto colon_pos = host.rfind(':');
   absl::string_view host_to_resolve = host;
   absl::optional<uint16_t> port;
+
+  if (host.empty()) {
```

Comment:
Hm, I see what you're saying about changing the precondition. If you want a minimal change, you could always keep the behavior of passing the empty string in to `parseInternetAddress` and it'll throw and return the same AuthorityAttributes. Just check that  `!potential_ip_address.empty()` before attempting to remove `[` and `]`. 

Want to fix the test array miscount too? Otherwise LGTM.

Reply:
Updated to your suggestion. I kept the test count the same but added an edge case explicitly.